lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -1171,3 +1171,11 @@ function dosomething_campaign_get_campaign_runs($node) {
 
   return $campaign_runs;
 }
+
+/**
+ * Implements hook_flush_caches()
+ *
+ */
+function dosomething_campaign_flush_caches() {
```

Comment:
you may want to add a flush_cache function for a specific campaign and update it on node updates